lib/locker/label.rb

```diff
@@ -75,9 +80,19 @@ def unlock!
             r.unlock!
           end
         end
+
+        if wait_queue.count > 0
+          next_user = wait_queue.shift
+          self.lock!(next_user)
+        end
```

Comment:
Instead of unlocking all the resources and then re-locking them, perhaps consider just transferring ownership to the next user? That would avoid any potential race conditions in the time between resources being unlocked and then locked. Pretty unlikely, I know. :)

Alternatively, since this is all redis, you could just wrap the whole thing in a MULTI block. http://redis.io/commands/multi